Jenkinsfile

```diff
@@ -0,0 +1,171 @@
+library 'magic-butler-catalogue'
+
+def PROJECT_NAME = "exclusive-lock"
+def CURRENT_BRANCH = [env.CHANGE_BRANCH, env.BRANCH_NAME]?.find{branch -> branch != null}
+def DEFAULT_BRANCH = 'main'
+def TRIGGER_PATTERN = ".*@logdnabot.*"
+
+pipeline {
+  agent {label 'ec2-fleet'}
+
+  options {
+    timestamps()
+    timeout time: 1, unit: 'HOURS'
+    ansiColor 'xterm'
+  }
+
+  triggers {
+    issueCommentTrigger(TRIGGER_PATTERN)
+  }
+
+  environment {
+    GITHUB_TOKEN = credentials('github-api-token')
+    NPM_CONFIG_CACHE = '.npm'
+    NPM_CONFIG_USERCONFIG = '.npmrc'
+    SPAWN_WRAP_SHIM_ROOT = '.npm'
+  }
+
+  stages {
+    stage('Validate PR Source') {
+      when {
+        expression { env.CHANGE_FORK }
+        not {
+          triggeredBy 'issueCommentCause'
+        }
+      }
+      steps {
+        error("A maintainer needs to approve this PR for CI by commenting")
+      }
+    }
+
+    stage('Test Suite') {
+      when {
+        beforeAgent true
+        not {
+          changelog '\\[skip ci\\]'
+        }
+      }
+
+      agent {
+        node {
+          label 'ec2-fleet'
+          customWorkspace "${PROJECT_NAME}-${BUILD_NUMBER}"
+        }
+      }
+
+      steps {
+        sh "mkdir -p ${NPM_CONFIG_CACHE}"
+        sh 'mkdir -p coverage'
+        script {
+          compose.up(
```

Comment:
We shouldn't use this anymore. just put the compose commands in package.json scrips and use npm run commands

Reply:
ok.  Why is groovy being phased out?

Reply:
This caused a chicken-and-egg problem. Can't run `npm` without a node image, and can't run `compose` with just a node image.  Gonna leave this since it works.